src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -20,29 +21,116 @@
     ConstraintTerm,
     BindingConstraint,
     ConstraintMatrixName,
+    BindingConstraintOperator,
+    BindingConstraintFrequency,
 )
 from antares.service.base_services import BaseBindingConstraintService
+from antares.tools.ini_tool import IniFile, IniFileTypes
+from antares.tools.time_series_tool import TimeSeriesFile, TimeSeriesFileType, TimeSeries
 
 
 class BindingConstraintLocalService(BaseBindingConstraintService):
     def __init__(self, config: LocalConfiguration, study_name: str, **kwargs: Any) -> None:
         super().__init__(**kwargs)
         self.config = config
         self.study_name = study_name
+        self._binding_constraints: dict[str, BindingConstraint] = {}
+        self.ini_file = IniFile(self.config.study_path, IniFileTypes.BINDING_CONSTRAINTS_INI)
+        self._time_series: dict[str, TimeSeries] = {}
 
     def create_binding_constraint(
         self,
         name: str,
         properties: Optional[BindingConstraintProperties] = None,
-        terms: Optional[List[ConstraintTerm]] = None,
+        terms: Optional[list[ConstraintTerm]] = None,
         less_term_matrix: Optional[pd.DataFrame] = None,
         equal_term_matrix: Optional[pd.DataFrame] = None,
         greater_term_matrix: Optional[pd.DataFrame] = None,
     ) -> BindingConstraint:
-        raise NotImplementedError
+        constraint = BindingConstraint(
+            name=name,
+            binding_constraint_service=self,
+            properties=properties,
+            terms=terms,
+        )
+        constraint.properties = constraint.local_properties.yield_binding_constraint_properties()
 
-    def add_constraint_terms(self, constraint: BindingConstraint, terms: List[ConstraintTerm]) -> List[ConstraintTerm]:
-        raise NotImplementedError
+        # Add binding constraints
+        self._binding_constraints[constraint.id] = constraint
+        self._write_binding_constraint_ini()
+
+        # Add constraint time series
+        time_series_length = (
+            (365 * 24 + 24) if constraint.properties.time_step == BindingConstraintFrequency.HOURLY else 366
+        )
+
+        if constraint.properties.operator in (BindingConstraintOperator.LESS, BindingConstraintOperator.BOTH):
```

Comment:
This if else logic is pretty long, perhaps you could place it inside a method to "hide" its complexity

Reply:
Yeah good call - should be extracted, it annoyed me when doing it too

Reply:
Extracted